common/web/lm-worker/src/main/correction/distance-modeler.ts

```diff
@@ -825,20 +784,24 @@ export class SearchSpace {
           // we can(?) assume that everything thereafter is as well.
           break;
         }
-        batch = batcher.checkAndAdd(newResult.finalNode);
-      }
 
-      if(batch) {
-        yield batch;
+        const entry = newResult.finalNode;
+
+        // As we can't guarantee a monotonically-increasing cost during the search -
+        // due to effects from keystrokes with deleteLeft > 0 - it's technically
+        // possible to find a lower-cost path later in such cases.
```

Comment:
This was actually observed with the setup established by existing unit tests:

https://github.com/keymanapp/keyman/blob/444ae921972f80b3d81bab3a75ffd8c186167f36/common/web/lm-worker/src/test/mocha/cases/worker-model-compositor.js#L798-L807

Trying to match something prefixed with "hi" is much lower-cost after the 'e' is deleted and the 'i' added in a single step; the conceptual "edge" from "he" to "hi" is thus negative-cost.

The assumption underlying the batching behavior was that path cost would always be monotonically-increasing... something that negative-cost edges invalidate.